cdap-ui/app/cdap/components/SelectionBox/index.tsx

```diff
@@ -63,54 +63,48 @@ export default function SelectionBox(props: ISelectionBoxProps) {
     );
   }, []);
 
-  React.useEffect(
-    () => {
-      if (!selection) {
-        return;
+  React.useEffect(() => {
+    if (!selection) {
+      return;
+    }
+    if (toggleSelection) {
+      selection.enable();
+    } else {
+      selection.disable();
+    }
+  }, [toggleSelection]);
+
+  React.useEffect(() => {
+    if (!selection) {
+      return;
+    }
+
+    selection.on('start', () => {
+      if (onSelectionStart) {
+        onSelectionStart();
       }
-      if (toggleSelection) {
-        selection.enable();
-      } else {
-        selection.disable();
+    });
+    selection.on('move', ({ selected, changed: { added, removed } }) => {
+      const selectedNodes = [];
+      for (const el of selected) {
+        selectedNodes.push(el.id);
       }
-    },
-    [toggleSelection]
-  );
-
-  React.useEffect(
-    () => {
-      if (!selection) {
-        return;
+      if (onSelectionMove && (added.length > 0 || removed.length > 0)) {
+        onSelectionMove({ selected: selectedNodes });
       }
+    });
+    selection.on('stop', () => {
+      if (onSelectionEnd) {
+        onSelectionEnd();
+      }
+    });
 
-      selection.on('start', () => {
-        if (onSelectionStart) {
-          onSelectionStart();
-        }
-      });
-      selection.on('move', ({ selected, changed: { added, removed } }) => {
-        const selectedNodes = [];
-        for (const el of selected) {
-          selectedNodes.push(el.id);
-        }
-        if (onSelectionMove && (added.length > 0 || removed.length > 0)) {
-          onSelectionMove({ selected: selectedNodes });
-        }
-      });
-      selection.on('stop', () => {
-        if (onSelectionEnd) {
-          onSelectionEnd();
-        }
-      });
-
-      return () => {
-        if (selection) {
-          selection.destroy();
-        }
-      };
-    },
-    [selection]
-  );
+    return () => {
+      if (selection) {
+        selection.destroy();
+      }
+    };
+  }, [selection]);
```

Comment:
does this need to be in another hook? Can this be clubbed with the on-mount hook that initializes the selection instance?

Reply:
It can be in the same hook. Just made it into two to make sure it is clear that one is onmount and the other is the regular hook listening on `selection`.